### PR TITLE
thing: interactor: Adapt schema type rules

### PR DESCRIPTION
**Pull request by alla3**
**Describe what this PR introduces:** (if necessary, link currently open issues with [special keywords](https://help.github.com/en/articles/closing-issues-using-keywords))

Adapt schema type rules to reflect the changes on commits [29523dd4d156473922ce55c09c14e9ac5c972269](https://github.com/CESARBR/knot-protocol-source/pull/9/commits/29523dd4d156473922ce55c09c14e9ac5c972269) and [9e9ea2b56a38af278e9fefd8eb5d7508ee61c0a6](https://github.com/CESARBR/knot-protocol-source/pull/9/commits/9e9ea2b56a38af278e9fefd8eb5d7508ee61c0a6) from [KNoT Protocol repository](https://github.com/CESARBR/knot-protocol-source). This patch should wait for those commits to be applied in the protocol repository.

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce any breaking changes?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] Related issues are referenced in the PR (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [ ] All tests are passing
- [ ] New/updated tests are included

**Testing environment:**

- Operating System/Platform:
- Go version:

If you have relevant logs, error output and etc, please attach them.

**Other information:**

### CODE REVIEW

**Review by esdrasjnr — CHANGES_REQUESTED** [review]
Looks great! :100: Just two comments:

- improve the commit message to make clear that the `valueType` is not associated with `typeId` anymore
- since this protocol change add some more value types, consider change the [data type checking](https://github.com/CESARBR/knot-babeltower/blob/b4a9de80f482495740733a5a0f71daf43a54201e/pkg/thing/interactors/update_data.go#L55) for `publishData` and `updateData` operations.

Thanks! :smile:

File: pkg/thing/interactors/update_schema.go [hunk 1]
```diff
@@ -19,33 +19,33 @@ type interval struct {
...
-	0x0012: {valueType: 2, unit: 1},              // FLOAT => LONGITUDE
-	0x0013: {valueType: 1, unit: interval{1, 4}}, // INT   => SPEED
-	0x0014: {valueType: 2, unit: interval{1, 6}}, // FLOAT => VOLUMEFLOW
-	0x0015: {valueType: 1, unit: interval{1, 6}}, // INT   => ENERGY
-	0xFFF0: {valueType: 3, unit: 0},              // BOOL  => PRESENCE
-	0xFFF1: {valueType: 3, unit: 0},              // BOOL  => SWITCH
-	0xFFF2: {valueType: 4, unit: 0},              // RAW   => COMMAND
-	0xFF10: {valueType: 1, unit: 0},              // INT   => ANALOG
-	0xFFFF: {valueType: 4, unit: 0},              // RAW   => INVALID
+	0x0000: {valueType: interval{1,7}, unit: 0},              // NONE
```
Comment (netoax) [inline pkg/thing/interactors/update_schema.go:22]: Which value types the interval 1-7 means?

Comment (netoax) [reply pkg/thing/interactors/update_schema.go:22]: I read your PR on the protocol repository and I got it. I'm updating this PR description to explicit it depends on that changes to be applied.

**Review by esdrasjnr — APPROVED** [review]
Excellent patch! Also, thanks for fixing the `float` typing check :100: 
Just consider improve the comments in `update_data.go` file by replacing the comments with `/* */` pattern! :+1:

File: pkg/thing/interactors/update_data.go [hunk 2]
```diff
@@ -73,3 +73,20 @@ func validateSchema(data entities.Data, schema []entities.Schema) bool {
 
 	return false
 }
+
+func validateSchemaNumber(value float64, valueType int) bool {
+	switch valueType {
+	case 1: /* int */
```
Comment (netoax) [inline pkg/thing/interactors/update_data.go]: considering it's a single line comment:

```suggestion
	case 1: // int
```

Comment (netoax) [reply pkg/thing/interactors/update_data.go]: same about the comments below.

**Review by netoax — CHANGES_REQUESTED** [review]
Very good, @alla3. Thanks for your contribution.
I made just one simple remark.

File: pkg/thing/interactors/update_data.go (hunk 2 above)
Comment (alla3) [reply pkg/thing/interactors/update_data.go]: Done.

**Review by netoax — APPROVED** [review]
LGTM!

**Pull request merged**